manifests/server/config.pp

```diff
@@ -201,15 +201,26 @@
       # - $service_name
       # - $port
       # - $datadir
+      file { 'systemd-conf-dir':
+        ensure => directory,
+        path   => "/etc/systemd/system/${service_name}.service.d",
+        owner   => root,
+        group   => root,
+        notify  => [ Exec['restart-systemd'], Class['postgresql::server::service'] ],
+        before  => Class['postgresql::server::reload'],
+      }
+
       file { 'systemd-override':
         ensure  => file,
-        path    => "/etc/systemd/system/${service_name}.service",
```

Comment:
For migrations it's a good idea to ensure the old location is absent. Otherwise an incorrect service could be loaded.

Reply:
You are correct.  I can add a resource to ensure that the old location is removed.

Reply:
I've added a resource to ensure the old file location is deleted.